dspace-api/src/test/java/org/dspace/content/RelationshipServiceImplTest.java

```diff
@@ -0,0 +1,287 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.content;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import org.dspace.authorize.service.AuthorizeService;
+import org.dspace.content.dao.RelationshipDAO;
+import org.dspace.content.service.ItemService;
+import org.dspace.content.virtual.VirtualMetadataPopulator;
+import org.dspace.core.Constants;
+import org.dspace.core.Context;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RelationshipServiceImplTest {
+
+    @InjectMocks
+    private RelationshipServiceImpl relationshipService;
+
+    @Mock
+    private RelationshipDAO relationshipDAO;
+
+    @Mock
+    private Context context;
+
+    @Mock
+    private Relationship relationship;
+
+    @Mock
+    private List<Relationship> relationshipsList;
+
+    @Mock
+    private AuthorizeService authorizeService;
+
+    @Mock
+    private ItemService itemService;
+
+    @Mock
+    private VirtualMetadataPopulator virtualMetadataPopulator;
+
+    @Before
+    public void init() {
+        relationshipsList = new ArrayList<>();
+        relationshipsList.add(relationship);
+    }
+
+    @Test
+    public void testFindAll() throws Exception {
+        when(relationshipDAO.findAll(context, Relationship.class)).thenReturn(relationshipsList);
+        assertEquals("TestFindAll 0", relationshipsList, relationshipService.findAll(context));
+    }
+
+    @Test
+    public void testFindByItem() throws Exception {
+        List<Relationship> relationshipTest = new ArrayList<>();
```

Comment:
Again inline comments would be helpful here. There's a lot of content & relationships getting mocked for this test.